crates/clvm-traits/src/clvm_decoder.rs

```diff
@@ -8,6 +8,35 @@ pub trait ClvmDecoder {
     fn decode_atom(&self, node: &Self::Node) -> Result<Atom, FromClvmError>;
     fn decode_pair(&self, node: &Self::Node) -> Result<(Self::Node, Self::Node), FromClvmError>;
 
+    fn decode_curried_arg(
+        &self,
+        node: &Self::Node,
+    ) -> Result<(Self::Node, Self::Node), FromClvmError> {
+        let (c, rest) = self.decode_pair(node)?;
```

Comment:
I would think you could decode this leveraging the existing `FromClvm` implementations for tuple and primitive as well.

Reply:
yeah that looks a lot better